### PR TITLE
cardano-cli: add governance create-info command

### LINKED ISSUE #289: missing create-info governance action
Current `main`

```
cardano-cli --version
cardano-cli 8.8.0.0 - linux-x86_64 - ghc-9.2
git rev 20b7b2b1e81cbb580736c93062be85485ac98ded
```

The command to create governance actions is missing the ability to: 

1.  create INFO governance actions necessary for SanchoNet

```
cardano-cli conway governance action
Usage: cardano-cli conway governance action
                                              ( create-constitution
                                              | create-new-committee
                                              | create-no-confidence
                                              | create-protocol-parameters-update
                                              | create-treasury-withdrawal
                                              )
```

### PR DESCRIPTION
**Pull request by smelc**
# Changelog

```yaml
- description: |
    Implementation of the create-info command
# uncomment types applicable to the change:
  type:
  - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Fixes https://github.com/input-output-hk/cardano-cli/issues/289

# Checklist

- [X] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [x] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [x] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] The changelog section in the PR is updated to describe the change
- [X] Self-reviewed the diff

### CODE REVIEW

**Comment by smelc** [comment]
As of , the output of `cardano-cli conway governance action create-info --help` is as follows:

```
Usage: cardano-cli conway governance action create-info (--mainnet | --testnet)
                                                          --governance-action-deposit NATURAL
                                                          ( --stake-pool-verification-key STRING
                                                          | --cold-verification-key-file FILE
                                                          | --stake-pool-id STAKE_POOL_ID
                                                          | --stake-verification-key STRING
                                                          | --stake-verification-key-file FILE
                                                          | --stake-key-hash HASH
                                                          )
                                                          --proposal-url TEXT
                                                          ( --proposal-text TEXT
                                                          | --proposal-file FILE
                                                          | --proposal-hash HASH
                                                          )
                                                          --out-file FILE

  Create an info action.

Available options:
  --mainnet                Use the mainnet magic id.
  --testnet                Use the testnet magic id.
  --governance-action-deposit NATURAL
                           Deposit required to submit a governance action.
  --stake-pool-verification-key STRING
                           Stake pool verification key (Bech32 or hex-encoded).
  --cold-verification-key-file FILE
                           Filepath of the stake pool verification key.
  --stake-pool-id STAKE_POOL_ID
                           Stake pool ID/verification key hash (either
                           Bech32-encoded or hex-encoded). Zero or more
                           occurences of this option is allowed.
  --stake-verification-key STRING
                           Stake verification key (Bech32 or hex-encoded).
  --stake-verification-key-file FILE
                           Filepath of the staking verification key.
  --stake-key-hash HASH    Stake verification key hash (hex-encoded).
  --proposal-url TEXT      Proposal anchor URL
  --proposal-text TEXT     Input proposal as UTF-8 encoded text.
  --proposal-file FILE     Input proposal as a text file.
  --proposal-hash HASH     Proposal anchor data hash.
  --out-file FILE          Path to action file to be used later on with build or
                           build-raw
  -h,--help                Show this help text
```

**Comment by newhoggy** [comment]
You will need to check in the `test/cardano-cli-golden/files/golden/help/conway_governance_action_create-info.cli` file as well.

File: cardano-cli/src/Cardano/CLI/EraBased/Commands/Governance/Actions.hs [hunk 1]
```diff
@@ -74,6 +74,19 @@ data NewConstitutionCmd
       , encFilePath :: File () Out
       } deriving Show
 
+-- | Datatype to carry data for the create-info governance action
+-- Will be modified later on to support passing the proposal as a file
+-- (so will be exclusive with the Url and HashSource)
+data NewInfoCmd
+   = NewInfoCmd
+      {   niNetwork :: Ledger.Network
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/EraBased/Commands/Governance/Actions.hs]: The alignment inside the curly braces is off. Have a look at: https://github.com/input-output-hk/cardano-node-wiki/wiki/Style-guide

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/Actions.hs [hunk 2]
```diff
@@ -2,7 +2,6 @@
 {-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE RankNTypes #-}
 {-# LANGUAGE TupleSections #-}
-{-# LANGUAGE TypeApplications #-}
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/Actions.hs:5]: :+1:

**Review by Jimbo4350 — APPROVED** [review]
Nice work :+1: . Just a small style issue to fix.

File: cardano-cli/src/Cardano/CLI/EraBased/Commands/Governance/Actions.hs (hunk 1 above)
Comment (smelc) [reply cardano-cli/src/Cardano/CLI/EraBased/Commands/Governance/Actions.hs]: Fixed :+1:

**Pull request merged**